SparkPost/2web2ui
# Sa-1528: Keep date range and filters when changing preset reports

### What Changed
 - Added preset tag to preset reports in config file.
 - Updated report options so that changing from unsaved report -> preset report & preset report -> preset report does not change date range and filters.
 - Added cypress tests for saved reports typeahead.

### How To Test
 - Go to `/signals/analytics` with Hibana enabled.
 - Add a filter ( I usually use campaign - Christmas sale)
 - Change the time range, timezone, and precision.
 - In the report field, select a preset report.
 - Verify that time range, timezone, precision, and filter does not get reset.
 - Select another preset report.
 - Verify again that time range, timezone, precision, and filter does not get reset.
